### PR TITLE
#340 clarify learning outcomes for week3

### PR DESCRIPTION
**Pull request by edgarfinn**
#340 

As discussed I've added: 

- How to implement a parallel functions.

- Introduction to the concept of and importance of software architecture: file structure planning, white-boarding an app, IIFEs and modular architecture

Any issues or suggestions from @lucyrose93 or @smarthutza please let me know 👍

### CODE REVIEW

**Comment by edgarfinn** [comment]
@lucyrose93 Done

File: coursebook/week-3/learning-outcomes.md [hunk 1]
```diff
@@ -18,4 +18,15 @@ You should be able to answer the following questions.
 ### ASYNC
 - What is a callback and why would you use one?
 - Why do we need asynchronous functionality in our code?
-- What do we mean by a "parallel function"? Why is this useful when querying APIs?
+- What do we mean by a "parallel function"?
+- Why are parallel functions useful when querying APIs?
+- How to implement a parallel function.
+
+### SOFTWARE ARCHITECTURE
+- Introduction to the concept and importance of software architecture.
```
Comment (jsms90) [inline coursebook/week-3/learning-outcomes.md]: You want this file to be a description of what students are supposed to walk away from the week knowing. All the points in this "Software Architecture" section sound less like "learning outcomes" and more like a description of the exercises that students are going to undertake.

File: coursebook/week-3/learning-outcomes.md [hunk 2]
```diff
@@ -18,4 +18,15 @@ You should be able to answer the following questions.
...
+- File structure planning and how white-board planning an app
```
Comment (jsms90) [inline coursebook/week-3/learning-outcomes.md]: Same here

File: coursebook/week-3/learning-outcomes.md [hunk 3]
```diff
@@ -18,4 +18,15 @@ You should be able to answer the following questions.
...
+- IIFEs and modular architecture
```
Comment (jsms90) [inline coursebook/week-3/learning-outcomes.md]: What should students know about IIFEs and modular architecture?

Comment (eliascodes) [inline coursebook/week-3/learning-outcomes.md]: Given the content of foundersandcoders/workshop-client-side-design#9, this should perhaps change to something like "understand the basic elements of software design in JavaScript (e.g. closures, abstraction, module pattern)

Comment (edgarfinn) [inline coursebook/week-3/learning-outcomes.md]: @lucyrose93 and @eliascodes would you be able to help me out with learning outcomes for the architecture and modular designs please? Struggling with this as was not so involved in those workshops / challenges. I'm happy for either of you to suggest wording for me to add to my PR. thanks

Comment (eliascodes) [inline coursebook/week-3/learning-outcomes.md]: "Understand the basic elements of software design in JavaScript, including closures, abstraction, module pattern."

**Comment by jsms90** [comment]
@edgarfinn This is kind of a work-around that doesn't really address the fact that the document now follows 2 different styles.

Given the current time constraints, I'm happy to merge this in and let this be something that maintainers pick up themselves. There are many areas of this repo that need some copy-editing for consistency of style.

@des-des @bradreeder @eliascodes please review

**Comment by eliascodes** [comment]
@jsms90 agree that it's a relatively minor consistency issue that shouldn't block merging at this stage. The content is basically fine.

@edgarfinn sorry to be annoying, but I didn't mean for the other two bullet points in the architecture section to be removed. The full list would be:
```
- The concept and importance of software architecture
- File structure planning and how white-board planning an app
- The basic elements of software design in JavaScript, including closures, abstraction, module pattern.
```

**Comment by edgarfinn** [comment]
@eliascodes ah ok, spotted a typo too so should be fixed now 👍

**Comment by eliascodes** [comment]
@edgarfinn thank you!

**Pull request merged**